src/java.base/share/classes/sun/security/util/AbstractAlgorithmConstraints.java

```diff
@@ -67,45 +69,37 @@ public String run() {
             }
             algorithmsInProperty = property.split(",");
             for (int i = 0; i < algorithmsInProperty.length; i++) {
-                algorithmsInProperty[i] = algorithmsInProperty[i].trim();
+                algorithmsInProperty[i] = algorithmsInProperty[i].trim().toLowerCase(Locale.ENGLISH);
```

Comment:
Is it possible to keep the current behavior that the property could be sensitive?  It may be not desired to allow "key size" in the property.

Reply:
If we keep property sensitive, we may need to use TreeSet. I have updated the PR with TreeSet. Fortunately, the performance hasn't changed much.

Reply:
I did not get the point to use TreeSet.  Is it sufficient if the toLowerCase() is not added (and don't compare keywords like "keySize" by ignoring cases)?

- algorithmsInProperty[i] = algorithmsInProperty[i].trim().toLowerCase(...);
+ algorithmsInProperty[i] = algorithmsInProperty[i].trim();

Reply:
Sorry, I missed a "case" in the original comment.  I meant to keep the property case sensitive in the hash set so that the keywords like "keySize" could be used correctly.

Reply:
[checkAlgorithm](https://github.com/openjdk/jdk/blob/a051e735cda0d5ee5cb6ce0738aa549a7319a28c/src/java.base/share/classes/sun/security/util/AbstractAlgorithmConstraints.java#L94) check whether the item is in the collection by ignoring case. If the item in the HashSet is case-sensitive, the method will lose its original algorithmic logic, but will retain it by using a ` new TreeSet<>(String.CASE_INSENSITIVE_ORDER);`

Can we use case sensitivity in checkAlgorithm to check an algorithm?

Reply:
The checkAlgorithm is using equalsIgnoreCase(), so it is safe for it.  My concern is mainly about the keywords, like "keySize" used the property, not really the algorithm name.  It is good to keep the current case sensitive checking behavior unchanged.

Reply:
Hi, Xuelei, thank you for your comments. I may not express it clearly, let me clarify. My concern is that if we use the HashSet:contains method to check whether an item is in the hash set, we cannot use equalsIgnoreCase(), so I used `new TreeSet<>(String.CASE_INSENSITIVE_ORDER)`  that can support equalsIgnoreCase().

According to my understanding, the current checkAlgorithm is not case sensitive, because it ignores the case of the item being checked. Looking forward to your suggestions。

Reply:
I see your point now.  Thank you for the clarification.  I need more time to think about if there is an improvement.  I will be back.